### PR TITLE
[#3145] Webform redirection

### PR DESCRIPTION
**Pull request by tangrammer**
Relates #3145 

#### Screenshots (if appropriate)
Screenshot 2020-05-27 at 11 18 40

Screenshot 2020-05-27 at 11 19 00

#### Reviewer Checklist
* [ ] Added an explanation about the work done
* [ ] Connected the PR and the issue on Zenhub
* [ ] Added a test plan to the issue
* [ ] Updated the copyright header (when relevant)
* [ ] Formatted the code
* [ ] Added a documentation (if relevant)
* [ ] Added some unit tests (if relevant)

### CODE REVIEW

File: GAE/src/org/akvo/flow/api/app/WebFormServlet.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        restRequest.populateFromHttpRequest(req);
+        return restRequest;
+    }
+
+    @Override
+    protected RestResponse handleRequest(RestRequest req) throws Exception {
+        RestResponse res = new RestResponse();
+        res.setCode(String.valueOf(HttpServletResponse.SC_OK));
+        String surveyId = ((WebFormRequest)req).getSurveyId();
+        String url = String.format("https://tech-consultancy.akvotest.org/akvo-flow-web/%s/%s", SystemProperty.applicationId.get(), surveyId);
```
Comment (tangrammer) [inline GAE/src/org/akvo/flow/api/app/WebFormServlet.java:54]: @muloem how can we read the flow env `akvotest` value in flow app? so we can use `prod` depending deployment environment

Comment (muloem) [reply GAE/src/org/akvo/flow/api/app/WebFormServlet.java:54]: You would need to add it to the appengine-web config and feed it in through the `EnvServlet.java`

Comment (tangrammer) [reply GAE/src/org/akvo/flow/api/app/WebFormServlet.java:54]: https://github.com/akvo/akvo-flow-server-config/pull/600

Comment (tangrammer) [reply GAE/src/org/akvo/flow/api/app/WebFormServlet.java]: https://github.com/akvo/akvo-flow/pull/3583/commits/2ef5a43c4c6154763a78398ce1e729b34699bb1e

**Comment by tangrammer** [comment]
@muloem I've included the webFormRedirectionUrl in akvo-flow-server-config https://github.com/akvo/akvo-flow-server-config/pull/600/files and commited changes to load the value ... could you take a look and correct them in case changes (or direction) are wrong?

File: GAE/src/org/akvo/flow/api/app/WebFormServlet.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        restRequest.populateFromHttpRequest(req);
+        return restRequest;
+    }
+
+    @Override
+    protected RestResponse handleRequest(RestRequest req) throws Exception {
+        RestResponse res = new RestResponse();
+        res.setCode(String.valueOf(HttpServletResponse.SC_OK));
+        String surveyId = ((WebFormRequest)req).getSurveyId();
+        String url = String.format("%s/%s/%s", PropertyUtil.getProperty(EnvServlet.WEBFORM_REDIRECTION_URL), SystemProperty.applicationId.get(), surveyId);
```
Comment (muloem) [inline GAE/src/org/akvo/flow/api/app/WebFormServlet.java:57]:
```suggestion
        String url = String.format("%s/%s/%s", PropertyUtil.getProperty(EnvServlet.WEBFORM_REDIRECTION_URL), PropertyUtil.getProperty("appId"), surveyId);
```

Comment (muloem) [reply GAE/src/org/akvo/flow/api/app/WebFormServlet.java:57]: We try to keep all property access through same class when possible.

File: GAE/src/org/akvo/flow/api/app/WebFormServlet.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    private static final Logger log = Logger.getLogger(WebFormServlet.class.getName());
+
+    public WebFormServlet() {
+        setMode(JSON_MODE);
+    }
+
+    @Override
+    protected RestRequest convertRequest() throws Exception {
+        HttpServletRequest req = getRequest();
+        log.warning(req.getPathInfo());
```
Comment (muloem) [inline GAE/src/org/akvo/flow/api/app/WebFormServlet.java:46]: Is this for debugging?

Comment (muloem) [reply GAE/src/org/akvo/flow/api/app/WebFormServlet.java:46]: if so maybe better `log.debug()` instead?

File: GAE/src/org/akvo/flow/api/app/WebFormServlet.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            sc = Integer.valueOf(resp.getCode());
+        } catch (NumberFormatException ignored) {
+            // Status code was not properly set in the RestResponse
+            sc = HttpServletResponse.SC_INTERNAL_SERVER_ERROR;
+        }
+        getResponse().setStatus(sc);
+        if (sc == HttpServletResponse.SC_OK) {
+            HttpServletResponse response = getResponse();
+            response.setContentType("text/html");
+            response.setStatus(response.SC_MOVED_TEMPORARILY);
```
Comment (muloem) [inline GAE/src/org/akvo/flow/api/app/WebFormServlet.java:79]: 👍

File: GAE/src/org/akvo/flow/api/app/WebFormServlet.java (hunk 3 above)
Comment (tangrammer) [reply GAE/src/org/akvo/flow/api/app/WebFormServlet.java:46]: forgot it, removing now

**Comment by tangrammer** [comment]
> I think its a good idea to explicitly add the `/webforms/*` point to the `webapp-security.xml` config. 👍

https://github.com/akvo/akvo-flow/pull/3583/commits/5fd34c082dc22697026357ce54ec41963e0f4920

**Pull request merged**